データベース設計

Naoki-Hashimoto opened the pull request:
#WHAT
README.mdにデータベースの概要を記載

#WHY
これから構築するデータベースの構造を明確にするため

Commit:
create pull request

Commit:
データベース設計

#WHAT
README.mdにデータベースの概要を記載

#WHY
これから構築するデータベースの構造を明確にするため

Commit:
データベース設計

#WHAT
README.mdにデータベースの概要を記載

#WHY
これから構築するデータベースの構造を明確にするため

madana-i commented on README.md:
```diff
@@ -22,3 +22,108 @@ Things you may want to cover:
 
 * ...
+
+***
```
デフォルトの記述は消しましょう！

madana-i commented on README.md:
```diff
@@ -22,3 +22,108 @@ Things you may want to cover:
+1. ログイン画面
+1. 新規登録画面
+  - 個別TOP画面
+        1. 新規gloup作成画面
```
groupですね

madana-i commented on README.md:
```diff
@@ -22,3 +22,108 @@ Things you may want to cover:
+###モデル間のリレーション
+####gloup & user
+  「多対多」であるため、groups_usersを媒介とし、
+  「has many 相手モデル名 through group_users」でアソシエーションを組む。
```
Railsで多対多のアソシエーションを組む方法は2通りあり、has_many throughは中間テーブルをモデルとしてRails内で使用する場合（中間部分にも独自の処理を持たせる場合）に使用するものです。
今回の場合、中間テーブルはテーブル同士のリレーション以外に使用する用途のみに使用するので、もう一方のアソシエーションが適切です。

madana-i commented on README.md:
```diff
@@ -22,3 +22,108 @@ Things you may want to cover:
+| 2 |サンプル乃介| test2@test.com |
+| 3 | Tech-man | test3@test.com |
+| 4 |   橋本   | test4@test.com |
+####group_user
```
groups_usersとgroup_userで表記揺れがあるので、正しいものに統一しましょう

Commit:
データベース設計2（デフォルトの記載を削除）

#WHAT
デフォルトの記載を削除

#WHY
これから構築するデータベース構造に関する記載のみを明確にしておくため

Commit:
データベース設計3（文言統一）

#WHAT
・“group”のスペルミス → 訂正
・groups_usersとgroup_userで表記揺れ
→ ”group_users”に統一

#WHY
誤解を招く表記であったため

Commit:
データベース設計4（リレーションに関する訂正）

#WHAT
groupとuserのアソセエーションを「has many through」で組むのでなく、
「has_and_belongs_to_many」で組む訂正。
#WHY
中間テーブルとなるgroup_usersで特別な処理を行う必要がないため。
